Do not replace error message for IO or format errors.

--- cryptsetup-1.0.6.old/lib/setup.c	2008-09-23 15:18:25.000000000 +0200
+++ cryptsetup-1.0.6/lib/setup.c	2008-09-23 15:25:45.000000000 +0200
@@ -484,13 +484,14 @@ start:
 	if(!password) {
 		r = -EINVAL; goto out;
 	}
-        
+
         r = LUKS_open_any_key(options->device, password, passwordLen, &hdr, &mk, backend);
-	if(r < 0) {
+	if (r == -EPERM)
 		set_error("No key available with this passphrase.\n");
+	if (r < 0)
 		goto out1;
-	} else
-                logger(options, CRYPT_LOG_NORMAL,"key slot %d unlocked.\n", r);
+
+	logger(options, CRYPT_LOG_NORMAL,"key slot %d unlocked.\n", r);
 
 	
 	options->offset = hdr.payloadOffset;
